homeassistant/components/light/__init__.py

```diff
@@ -1060,6 +1061,49 @@ def _light_internal_convert_color(
             data[ATTR_XY_COLOR] = color_util.color_hs_to_xy(*hs_color)
         return data
 
+    def __validate_color_mode(
+        self,
+        color_mode: ColorMode | str | None,
+        supported_color_modes: set[ColorMode] | set[str],
+        effect: str | None,
+    ) -> None:
+        """Validate the color mode."""
+        if color_mode is None:
+            # The light is turned off
+            return
+
+        if not effect or effect == EFFECT_OFF:
+            # No effect is active, the light must set color mode to one of the supported
+            # color modes
+            if color_mode in supported_color_modes:
+                return
+            # Increase severity to warning in 2024.2, reject in 2025.2
+            _LOGGER.debug(
+                "%s: set to unsupported color_mode: %s, supported_color_modes: %s",
+                self.entity_id,
+                color_mode,
+                supported_color_modes,
+            )
+            return
+
+        # When an effect is active, we allow the light to set its color mode to on_off and
+        # to brightness if the light allows adjusting brightness
+        effect_color_modes = supported_color_modes | {ColorMode.ONOFF}
```

Comment:
I don't understand this. This will allow all supported color modes and color mode onoff. How do we check that the light has limited its color mode with this?

Reply:
The intention is to allow any color mode supported by the light, *and in addition*, also `ColorMode.ONOFF`

The implementation is correct (I think), but the comment is unclear. I'll try to improve it.